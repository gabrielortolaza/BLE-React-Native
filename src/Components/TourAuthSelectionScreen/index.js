/** Page appears to no longer be used */
/* eslint-disable */
import React, { PureComponent }  from "react";
import PropTypes from "prop-types";
import { SafeAreaView } from "react-native-safe-area-context";
import { ActivityIndicator, View } from "react-native";

import StyleSheet from "../../Proportional";
import { Styles, Colors } from "../../Themes";
import { RoundButton, BackButton, Welcome } from "./../Shared";

const emptyFunction = () => {};

class TourAuthSelectionScreen extends PureComponent {
  render() {
    const {
      goBack, goToSignIn, goToSignUp,
      signUpWithFacebook, isWaitingFacebook
    } = this.props;

    return (
      <SafeAreaView style={styles.container}>
        <View testID="TourAuthSelectionScreen" style={styles.container}>
          <BackButton testID="TourAuthSelection_Back" onPress={goBack} />
          <Welcome
            testID="TourAuthSelection_Welcome"
            title="Nice!"
            subtitle="Time to make your account"
            alt
          />
          <RoundButton
            alt
            facebook
            onPress={isWaitingFacebook ? emptyFunction : signUpWithFacebook}
            override={isWaitingFacebook}
            positionStyle={styles.facebookPositionStyle}
            testID="TourAuthSelection_Facebook"
          >
            {isWaitingFacebook ? (
              <ActivityIndicator
                testID={"TourAuthSelection_FBActivity"}
                color="white"
              />
            ) : (
              "Continue With Facebook"
            )}
          </RoundButton>
          <RoundButton
            onPress={goToSignUp}
            positionStyle={styles.signUpPositionStyle}
            wide
            testID="TourAuthSelection_SignUp"
          >
            Sign up with email
          </RoundButton>
          <RoundButton
            onPress={goToSignIn}
            positionStyle={styles.signInPositionStyle}
            white
            wide
            testID="TourAuthSelection_SignIn"
          >
            I already have an account
          </RoundButton>
        </View>
      </SafeAreaView>
    );
  }
}

export default TourAuthSelectionScreen;

TourAuthSelectionScreen.propTypes = {
  goBack: PropTypes.func.isRequired,
  goToSignIn: PropTypes.func.isRequired,
  goToSignUp: PropTypes.func.isRequired,
  isWaitingFacebook: PropTypes.bool,
  signUpWithFacebook: PropTypes.func.isRequired
};

const styles = StyleSheet.createProportional({
  container: {
    flex: 1,
    justifyContent: "center",
    backgroundColor: Colors.white
  },
  input: {
    marginTop: 14,
    marginHorizontal: 58
  },
  image: {
    alignSelf: "center"
  },
  tipLabel: {
    ...Styles.TipLabel1,
    color: Colors.white,
    textAlign: "center",
    marginTop: 3
  },
  backButton: {
    position: "absolute",
    top: 30,
    padding: 10,
    left: 20,
    zIndex: 10
  },
  facebookPositionStyle: {
    marginTop: 15
  },
  signUpPositionStyle: {
    marginTop: 13
  },
  signInPositionStyle: {
    marginTop: 46
  },
  indicatorWrapper: {
    height: 40,
    justifyContent: "center"
  }
});
