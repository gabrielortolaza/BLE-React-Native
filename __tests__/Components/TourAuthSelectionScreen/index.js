/** Page appears to no longer be used */

import React from "react";
import PropTypes from "prop-types";
import { shallow } from "enzyme";
import sinon from "sinon";

import TourAuthSelectionScreen from "../../../src/Components/TourAuthSelectionScreen";

describe("TourAuthSelectionScreen", () => {
  it("Default", () => {
    const goBack = sinon.spy(jest.fn());
    const goToSignIn = sinon.spy(jest.fn());
    const goToSignUp = sinon.spy(jest.fn());
    // const signUpWithFacebook = sinon.spy(jest.fn());

    const wrapper = shallow(
      <TourAuthSelectionScreen
        goBack={goBack}
        goToSignIn={goToSignIn}
        goToSignUp={goToSignUp}
      />
    );

    const render = wrapper.dive();

    expect(render.find({ testID: "TourAuthSelection_Welcome" })).toHaveLength(1);

    render.find({ testID: "TourAuthSelection_Back" }).simulate("press");
    expect(goBack.calledOnce).toBe(true);
    expect(goBack.callCount).toBe(1);

    expect(render.find({ testID: "TourAuthSelection_FBActivity" })).toHaveLength(0);

    render.find({ testID: "TourAuthSelection_SignUp" }).simulate("press");
    expect(goToSignUp.calledOnce).toBe(true);

    render.find({ testID: "TourAuthSelection_SignIn" }).simulate("press");
    expect(goToSignIn.calledOnce).toBe(true);

    render.find({ testID: "TourAuthSelection_SignUp" }).simulate("press");
    render.find({ testID: "TourAuthSelection_SignUp" }).simulate("press");

    render.find({ testID: "TourAuthSelection_SignIn" }).simulate("press");
    render.find({ testID: "TourAuthSelection_SignIn" }).simulate("press");
    render.find({ testID: "TourAuthSelection_SignIn" }).simulate("press");

    expect(goToSignIn.callCount).toBe(4);
    expect(goToSignUp.callCount).toBe(3);

    expect(render).toMatchSnapshot();
  });

  //   it('Facebook Waiting', () => {
  //     const goBack = sinon.spy(jest.fn())
  //     const goToSignIn = sinon.spy(jest.fn())
  //     const goToSignUp = sinon.spy(jest.fn())
  //     const signUpWithFacebook = sinon.spy(jest.fn())

  //     const wrapper = shallow(
  //       <TourAuthSelectionScreen
  //         goBack={goBack}
  //         goToSignIn={goToSignIn}
  //         goToSignUp={goToSignUp}
  //         isWaitingFacebook
  //         signUpWithFacebook={signUpWithFacebook}
  //       />
  //     )

  //     const render = wrapper.dive()

  //     expect(render.find({ testID: 'TourAuthSelection_FBActivity' })).toHaveLength(1)

  //     render.find({ testID: 'TourAuthSelection_Facebook' }).simulate('press')
  //     expect(signUpWithFacebook.calledOnce).toBe(false)

  //     expect(render).toMatchSnapshot()
  //   });
});

TourAuthSelectionScreen.propTypes = {
  goBack: PropTypes.func.isRequired,
  goToSignIn: PropTypes.func.isRequired,
  goToSignUp: PropTypes.func.isRequired,
  isWaitingFacebook: PropTypes.bool
};
