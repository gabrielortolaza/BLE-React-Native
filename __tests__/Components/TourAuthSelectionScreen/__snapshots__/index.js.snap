// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourAuthSelectionScreen Default 1`] = `ShallowWrapper {}`;
